libs/clients/gemini/client.go

```diff
@@ -396,9 +396,6 @@ func (c *HTTPClient) CheckTxStatus(ctx context.Context, APIKey string, clientID
 
 	var body PayoutResult
 	resp, err := c.client.Do(ctx, req, &body)
```

Comment:
I would do something like this instead.  The error returned can be an errorbundle which includes the http state of the response on it.  If you do this and you get a connection timeout or something the code will panic about nil response.

```go
        _, err = c.client.Do(ctx, req, &body)                                                                                            
        if err != nil {                                                                                                                  
                var eb *errorutils.ErrorBundle                                                                                           
                if errors.As(err, eb) {                                                                                                  
                        if httpState, ok := eb.Data().(clients.HTTPState); ok {                                                          
                                if httpState.Status == http.StatusNotFound {                                                             
                                        notFoundReason := "404 From Gemini"                                                              
                                        return &PayoutResult{                                                                            
                                                Result: "Error",                                                                         
                                                Reason: &notFoundReason,                                                                 
                                                TxRef:  txRef,                                                                           
                                        }, nil                                                                                           
                                }                                                                                                        
                        }                                                                                                                
                }                                                                                                                        
                return nil, err                                                                                                          
        }
```

Reply:
OK done.

Reply:
maybe @Sneagan too

Reply:
looking